app/(candidate)/dashboard/purchase/components/PurchasePage.js

```diff
@@ -0,0 +1,67 @@
+'use client'
+
+import { useState } from 'react'
+import { completePurchase } from '../utils/purchaseFetch.utils'
+import PurchaseError from './PurchaseError'
+import PurchaseSuccess from './PurchaseSuccess'
+import PurchasePayment from './PurchasePayment'
+
+export default function PurchasePage({
+  type,
+  domain,
+  websiteId,
+  returnUrl,
+  purchaseIntent,
+  error: serverError,
+}) {
+  const [purchaseState, setPurchaseState] = useState('payment')
+  const [error, setError] = useState(null)
+
+  const handlePaymentSuccess = async (paymentIntent) => {
+    try {
+      const response = await completePurchase(paymentIntent.id)
+
+      if (response.ok) {
+        setPurchaseState('success')
+      } else {
+        setError(response.data?.data?.error || 'Failed to complete purchase')
+        setPurchaseState('error')
+      }
+    } catch (error) {
+      setError('Failed to complete purchase')
+      setPurchaseState('error')
+    }
+  }
+
+  const handlePaymentError = (error) => {
+    setError(error.message)
+    setPurchaseState('error')
```

Comment:
Can we please add these `purchaseState` string literals to an enumerated const somewhere? 🙏